streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -855,6 +856,20 @@ public void advanceWallClockTime(final Duration advance) {
         return new TestOutputTopic<>(this, topicName, keyDeserializer, valueDeserializer);
     }
 
+    /**
+     * Get all the names of all the topics to which records have been output.
+     * <p>
+     * Call this method after piping the input into the test driver to retrieve the full set of topics the topology
+     * produced records to.
+     * <p>
+     * The returned set of topic names includes changelog, repartition and sink topic names.
+     *
+     * @return the set of output topic names.
+     */
+    public final Set<String> getOutputTopicNames() {
```

Comment:
@big-andy-coates Can you update the method name according to the KIP discussion?